### PR TITLE
Disallow TLS specific args for non-TLS

### PR DESCRIPTION
**Pull request by unparalleled-js**
* Fixes issue where the user was still allowed to pass in `--certs` and `--ignore-cert-validation` when not using the `TLS-TCP` protocol for `send-to` commands.

### CODE REVIEW

File: src/code42cli/cmds/search/__init__.py [hunk 1]
```diff
@@ -21,10 +22,29 @@ def invoke(self, ctx):
...
             hostname, protocol, output_format, certs
         )
         return super().invoke(ctx)
+
+
+def _handle_incompatible_args(protocol, ignore_cert_validation, certs):
+    if protocol == ServerProtocol.TLS_TCP:
+        return
+
+    arg = (
```
Comment (alanag13) [inline src/code42cli/cmds/search/__init__.py:40]: Nested ternaries are a little hard to parse, could we just use if blocks?

File: tests/cmds/test_alerts.py [hunk 2]
```diff
@@ -789,6 +789,52 @@ def test_send_to_when_given_ignore_cert_validation_uses_certs_equal_to_ignore_st
     )
 
 
+@pytest.mark.parametrize("protocol", (ServerProtocol.UDP, ServerProtocol.TCP))
```
Comment (alanag13) [inline tests/cmds/test_alerts.py:792]: These tests are good examples of when parameterized tests make sense.

**Comment by timabrmsn** [comment]
Oh hey, I had started a branch to add a generic `depends_on` option class factory like `incompatible_with` not too long ago, but since we didn't have a need for it yet I hadn't wrapped it up. This seems like a perfect time to add that.

**Comment by timabrmsn** [comment]
Actually, my `depends_on` won't work as is here, since we're depending on the specific value of an option, not just the presence of it..

File: src/code42cli/cmds/search/__init__.py (hunk 1 above)
Comment (unparalleled-js) [reply src/code42cli/cmds/search/__init__.py:40]: I did that originally but felt the opposite. I can put it back and show you what I mean

File: tests/cmds/test_alerts.py (hunk 2 above)
Comment (unparalleled-js) [reply tests/cmds/test_alerts.py:792]: Right!

File: src/code42cli/cmds/search/__init__.py [hunk 3]
```diff
@@ -21,10 +22,29 @@ def invoke(self, ctx):
...
+        "--ignore-cert-validation"
+        if ignore_cert_validation is not None
+        else "--certs"
+        if certs is not None
+        else None
+    )
+    if arg is not None:
+        raise Code42CLIError(
```
Comment (timabrmsn) [inline src/code42cli/cmds/search/__init__.py:48]: This should instead raise `click.BadOptionUsage`.

File: src/code42cli/cmds/search/__init__.py (hunk 1 above)
Comment (unparalleled-js) [reply src/code42cli/cmds/search/__init__.py:40]: Ah I guess it's not too bad, nevermind

**Comment by unparalleled-js** [comment]
> Actually, my `depends_on` won't work as is here, since we're depending on the specific value of an option, not just the presence of it..

That seems like it would work?  

`--certs` depends on a specific value of `--protocol`, etc

**Review by timabrmsn — APPROVED** [review]
